arangod/Aql/Query.cpp

```diff
@@ -600,7 +599,10 @@ ExecutionState Query::execute(QueryResult& queryResult) {
         }
         // will set warnings, stats, profile and cleanup plan and engine
         auto state = finalize(*queryResult.extra);
-        if (state == ExecutionState::DONE && _cacheEntry != nullptr) {
+        if (state == ExecutionState::DONE && _cacheEntry != nullptr &&
+            !((_transactionContext->isStreaming() ||
+               _transactionContext->isTransactionJS()) &&
+              !_trx->state()->isReadOnlyTransaction())) {
```

Comment:
This `if` condition is mind-blowing. double negation :scream: 
Can it be written in a simpler way?